### PR TITLE
Ergonomics

### PR DESCRIPTION
**Pull request by QnnOkabayashi**
This PR brings way too many things because I'm too lazy to be fancy with git.

* Changed `fail::*` panicking functions to `&str`s so that `expect` can be used instead of `unwrap_or_else`.
* Gave `Processor` an explicit error type created by `processor::error`.
* Rename `tree::Span`'s `children` field to `nodes`.
* Changed the `Uuid` parser to return `Option<Uuid>` instead of `Result<Uuid, ()>`.
* `tree::Event` now holds an `Option<Tag>` instead of a `Tag`, clarifying if a tag was explicitly provided.
* Renamed `builder` module to `runtime`, and `builder::LayerBuilder` to `runtime::Builder`.
* Improved documentation of everything (fixed typos, added examples, etc.)
* Added a `processor::from_fn` method to create `Processor`s without having to explicitly create new types.
  * Removed `Processor` impls for `std` channel sender types and `tokio::sync::mpsc::UnboundedSender`.
* Added `Tag::build` method, allowing for clearer configuration of `Tag` values.
  * Removed `tag!` macro.
* Renamed `TagParser::try_parse` to `TagParser::parse`.

Hopefully I'm not forgetting anything :)

EDIT: Reverted `Tag` build pattern using closures to the regular build pattern [because of this](https://github.com/idanarye/rust-typed-builder/issues/66#issuecomment-1114353926). This removes
```rust
Tag::build(|builder| builder.level(...))
```
and replaces it with
```rust
Tag::builder().level(...).build()
```

### CODE REVIEW

**Review by Firstyear — APPROVED** [review]

**Comment by Firstyear** [comment]
Looks awesome!

File: tracing-forest/src/fail.rs [hunk 1]
```diff
@@ -1,20 +1,14 @@
...
+pub const SPAN_NOT_IN_CONTEXT: &str = "Span not in context, this is a bug";
+pub const OPENED_SPAN_NOT_IN_EXTENSIONS: &str =
+    "Span extension doesn't contain `OpenedSpan`, this is a bug";
+pub const PROCESSING_ERROR: &str = "Processing logs failed";
 
 cfg_uuid! {
+    pub const NO_CURRENT_SPAN: &str = "The subscriber isn't in any spans";
+    pub const NO_FOREST_LAYER: &str = "The span has no `Span` in extensions, perhaps you forgot to add a `ForestLayer` to your subscriber?";
+
     #[cold]
```
[the remark this answers is not in the document]
Comment (QnnOkabayashi) [reply tracing-forest/src/fail.rs:12]: I think it tells the optimizer to assume it won't be called very much. idk I saw it in source code in std somewhere and it compiles fine sooo...

File: tracing-forest/src/runtime.rs [hunk 2]
```diff
@@ -100,26 +100,25 @@
...
 use std::future::Future;
-use std::iter::from_fn;
-use tokio::sync::mpsc::{self, UnboundedReceiver, UnboundedSender};
+use std::iter;
+use tokio::sync::mpsc::{self, UnboundedReceiver};
 use tokio::sync::oneshot;
 use tracing::Subscriber;
 use tracing_subscriber::Registry;
-use tracing_subscriber::layer::{Layer, Layered};
+use tracing_subscriber::layer::{Layered, SubscriberExt as _};
```
[the remark this answers is not in the document]
Comment (QnnOkabayashi) [reply tracing-forest/src/runtime.rs:116]: Brings the trait into scope without polluting the namespace, so you can call methods required by that trait. Same as what happens in `tracing_forest::traits`.

[the remark this answers is not in the document]
Comment (Firstyear) [reply tracing-forest/src/runtime.rs:116]: Sick, thanks for explaining!

**Pull request merged**